src/main/java/jenkins/plugins/office365connector/FactsBuilder.java

```diff
@@ -0,0 +1,169 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package jenkins.plugins.office365connector;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import hudson.model.Cause;
+import hudson.model.Result;
+import hudson.model.Run;
+import hudson.model.User;
+import hudson.scm.ChangeLogSet;
+import hudson.tasks.test.AbstractTestResultAction;
+import jenkins.plugins.office365connector.model.Fact;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * Collects helper methods that create instance of {@link jenkins.plugins.office365connector.model.Fact Fact} class.
+ *
+ * @author Damian Szczepanik (damianszczepanik@github)
+ */
+public class FactsBuilder {
+
+    final static String NAME_STATUS = "Status";
+    final static String NAME_REMARKS = "Remarks";
+    final static String NAME_CULPRITS = "Culprits";
+    final static String NAME_DEVELOPERS = "Developers";
+    final static String NAME_NUMBER_OF_CHANGED_FILES = "Number of files changed";
+
+    final static String NAME_START_TIME = "Start time";
+    final static String NAME_COMPLETION_TIME = "Completion time";
+    final static String NAME_BACK_TO_NORMAL_TIME = "Back to normal time";
+    final static String NAME_FAILING_SINCE_TIME = "Failing since time";
+    final static String NAME_FAILING_SINCE_BUILD = "Failing since build";
+
+    final static String NAME_TOTAL_TESTS = "Total tests";
+    final static String NAME_FAILED_TESTS = "Failed tests";
+    final static String NAME_SKIPPED_TESTS = "Skipped tests";
+    final static String NAME_PASSED_TESTS = "Passed tests";
+
+    private final List<Fact> facts = new ArrayList<>();
+
+    public void addStatusStarted() {
+        facts.add(new Fact(NAME_STATUS, "Started"));
+    }
+
+    public void addStatusRunning() {
+        facts.add(new Fact(NAME_STATUS, "Running"));
+    }
+
+    public static Fact buildStatus() {
+        return new Fact(NAME_STATUS);
+    }
+
+    public void addStartTime(Run run) {
+        facts.add(new Fact(NAME_START_TIME, TimeUtils.dateToString(run.getStartTimeInMillis())));
```

Comment:
If the ``run`` is ``null`` then this method will never be reached because of the NPE thrown earlier. And as I checked the code had no NPT validation so I guess there is no need for this. So if the ``run`` is ``null`` then the plugin failed already and the source is not here.